From 6df12d8b33a86f91226c883a4239ebabf9ba5d6b Mon Sep 17 00:00:00 2001
From: Orlando Vazquez <ovazquez@gmail.com>
Date: Wed, 29 May 2019 00:15:40 -0700
Subject: [PATCH] TRITON-1049 Migration (RFD34) limit values should be stored
 inside SAPI Reviewed by: Todd Whiteman <todd.whiteman@joyent.com> Approved
 by: Todd Whiteman <todd.whiteman@joyent.com>

---
 docs/index.md                 | 1 +
 lib/apis/wfapi.js             | 1 +
 package.json                  | 2 +-
 sapi_manifests/vmapi/template | 3 +++
 server.js                     | 1 +
 5 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/docs/index.md b/docs/index.md
index 89871f7..2073351 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -2661,6 +2661,7 @@ production.
 | ------------------------------ | ------ | ---------------------------------------------------------------------------- |
 | **experimental_fluentd_host**  | String |                                                                              |
 | **docker_tag_re**              | String | Tags matching regex are treated with Docker tag semantics                    |
+| **migration_send_mbps_limit**  | Number | Limit of transfer rate, in megabits per second, for individual migrations    |
 
 `docker_tag_re` must be a valid regular expression string -- more concretely,
 what Javascript's RegExp() considers valid. Docker tags can be added during
diff --git a/lib/apis/wfapi.js b/lib/apis/wfapi.js
index bbcebc4..7688bf9 100644
--- a/lib/apis/wfapi.js
+++ b/lib/apis/wfapi.js
@@ -813,6 +813,7 @@ Wfapi.prototype.createMigrateJob = function _createMigrateJobFn(req, cb) {
         override_alias: req.params.override_alias,
         server_uuid: req.vm.server_uuid, // Used by the acquireVMTicket calls.
         target: '/migrate-' + vm_uuid,
+        zfs_send_mbps_limit: req.app.options.zfs_send_mbps_limit,
         task: workflowTask,
         vm: req.vm,
         vm_uuid: vm_uuid,
diff --git a/package.json b/package.json
index 90e7416..d4ad1b3 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.8.7",
+  "version": "9.8.8",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
diff --git a/sapi_manifests/vmapi/template b/sapi_manifests/vmapi/template
index 139ea48..72e2630 100644
--- a/sapi_manifests/vmapi/template
+++ b/sapi_manifests/vmapi/template
@@ -27,6 +27,9 @@
         "enabled": false
 {{/fabric_cfg}}
     },
+{{#migration_send_mbps_limit}}
+    "migration_send_mbps_limit": {{migration_send_mbps_limit}},
+{{/migration_send_mbps_limit}}
     "wfapi": {
         "forceMd5Check": true,
         "workflows": [
diff --git a/server.js b/server.js
index f2b367f..e28a1a3 100644
--- a/server.js
+++ b/server.js
@@ -303,6 +303,7 @@ function startVmapiService() {
                 metricsManager: metricsManager,
                 moray: moray,
                 morayBucketsInitializer: morayBucketsInitializer,
+                zfs_send_mbps_limit: config.migration_send_mbps_limit,
                 overlay: config.overlay,
                 cnapi: config.cnapi, // TODO: TRITON-1295
                 reserveKvmStorage: config.reserveKvmStorage,
-- 
2.21.0

